### DISCUSSION TITLE
Interactive online experiments

### DISCUSSION THREAD

**Discussion by MelindaPozzi** [jspsych/jsPsych#779] [Q&A] [answered]
Hi,

Is it possible to design interactive online experiments (where participants interact with other participants) with jsPsych? Is there the possibility to have a chat system that participants can use to communicate? And is it also possible to design interactive online experiments where the participant plays with a virtual partner (the computer)?

Thank you in advance for your answer!

**Comment by vijaymarupudi** [comment, marked as answer]
Yep, those are all possible via Javascript, jsPsych is just a thin layer over the programming language.

For chat applications, look into the Websocket API or the long polling technique. You will need to write a corresponding server (if it's person to person communication). 

It is also possible to design interactive experiments. You would need to write all the logic for the virtual partner. There are certain Javascript libraries like Phaser.js to help with that.

Let me know if you have any further questions!

\~ Vijay

**Comment by MelindaPozzi** [comment]
Hi,

Thank you very much for your answer! I will come back if I have further questions.

Best,
Mélinda

**Reply by becky-gilbert** [reply]
Hi,
Just wanted to mention that JATOS is set up to do this, using what they call "group studies". JATOS is a free and open source server-side program for managing online experiments. I believe they have a demo of an experiment that allows participants to chat with each other. See here: http://www.jatos.org/Example-Group-Studies.html
Becky

**Reply by MelindaPozzi** [reply]
Hi,
Thank you very much for this information! I will look at it.
Mélinda
